pkg/server/datastore/sqlstore/models.go

```diff
@@ -123,6 +123,37 @@ func (DNSName) TableName() string {
 	return "dns_names"
 }
 
+// FederatedTrustDomain holds federated trust domains.
+// It has the information needed to get updated bundles of the
+// federated trust domain from a SPIFFE bundle endpoint server.
+type FederatedTrustDomain struct {
+	Model
+
+	// TrustDomain is the trust domain name (e.g., "example.org") to federate with.
+	TrustDomain string `gorm:"not null;unique_index"`
+
+	// BundleEndpointURL is the URL of the SPIFFE bundle endpoint that provides the trust
+	// bundle to federate with.
+	BundleEndpointURL string
+
+	// BundleEndpointProfile is the endpoint profile type.
+	BundleEndpointProfile string
+
+	// EndpointSPIFFEID specifies the expected SPIFFE ID of the
+	// SPIFFE bundle endpoint server when BundleEndpointProfile
+	// is "https_spiffe"
+	EndpointSPIFFEID string
+
+	// FederateWithNewRegistrationEntries indicates wether the trust domain
+	// automatically federates with new registration entries by default or not.
+	FederateWithNewRegistrationEntries bool
```

Comment:
This implies that existing registration entries won't federate with this trust domain. @evan2645 can correct me if I'm wrong, but I _think_ the behavior we want is that if this boolean is set, there is an implied `FederatesWith` for _all_ registration entries. I think the implementation of this feature would not involve actually changing the FederatesWith list for any entries when they are stored, but rather adding these trust domains to the list on egress.
If so, maybe a better name would be something simple like `Implicit bool`.

Reply:
I was not sure if the intention was to include existing registration entries in this behavior or not. I had some doubts about it but now that I think it more it makes sense.
I renamed the field to the suggested `implicit` name, which looks a lot better :)